### ISSUE TITLE
Allow `AdvancedSubscriber` to detect `AdvancedPublisher` going away

### ISSUE THREAD

**Issue by skreborn** [eclipse-zenoh/zenoh#1669]
While `AdvancedSubscriber::detect_publishers` allows to manually subscribe to the liveliness of an `AdvancedPublisher`, this requires the user to keep around and manually synchronize the incoming messages from both.

**Comment by OlivierHecart** [comment]
If I understood well the issue. You would like to receive both data and liveliness events in the same callback or stream. This is something you can already do on top of Zenoh. 
The problem with this is that you need to change the type of 'event' received by the callback or stream to a type (enum typically) that can handle both data and liveliness. This would dealign Subscriber and AdvancedSubscriber APIs and complexify the AdvancedSubscriber API for every user (even ones that don't care about liveliness). For those reasons we will probably not do this in Zenoh API.

**Comment by skreborn** [comment]
Indeed, the increased API complexity by introducing an enumerated type is a perfectly valid reason not to add this directly into the callback/stream. While I personally wouldn't mind the dissonance between the plain `Subscriber` and the `AdvancedSubscriber`, I understand that the less differences are, the easier it is for developers to understand the library and use it correctly.
